flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/exceptionhistory/FailureHandlingResultSnapshot.java

```diff
@@ -59,38 +58,60 @@
     public static FailureHandlingResultSnapshot create(
             FailureHandlingResult failureHandlingResult,
             Function<ExecutionVertexID, Execution> latestExecutionLookup) {
+        return create(
+                failureHandlingResult.getExecutionVertexIdOfFailedTask().orElse(null),
+                failureHandlingResult.getError(),
+                failureHandlingResult.getVerticesToRestart(),
+                failureHandlingResult.getTimestamp(),
+                latestExecutionLookup);
+    }
+
+    /**
+     * Creates a {@code FailureHandlingResultSnapshot} based on passed parameters.
+     *
+     * @param failingExecutionVertexId an {@link Optional} of the {@link ExecutionVertexID} the
+     *     failure originates from, or {@code None}.
```

Comment:
The JavaDoc would need to be updated as well since it's a `@Nullable` object instead of an `Optional` now.